src/vmm/src/devices/virtio/queue.rs

```diff
@@ -443,12 +443,11 @@ impl Queue {
     ) -> Result<(), QueueError> {
         debug_assert!(self.is_layout_valid(mem));
 
-        let next_used = self.next_used.0 % self.actual_size();
         let used_element = UsedElement {
             id: u32::from(desc_index),
             len,
         };
-        self.write_used_ring(mem, next_used, used_element)?;
```

Comment:
Conceptually I understand that, but `write_used_ring` is a private function of this module which is only called by `add_used`. So, why change it? What use-case do you foresee that might break things because a user did not calculate properly the actual index?

According to your current commit message and the actual diff, this change does not add any value, it's just aesthetics. If there is a future use-case please mention it in the commit message. If not, please drop the commit.

Reply:
ok, I removed commit from this PR.